src/osdc/Objecter.cc

```diff
@@ -2460,6 +2460,10 @@ void Objecter::_op_submit(Op *op, shunique_lock& sul, ceph_tid_t *ptid)
   sl.unlock();
   put_session(s);
 
+ if ((cct->_conf->rgw_fastfail_homeless_timeout > 0) && s->is_homeless()) { 
+     uint64_t t = timer.add_event(ceph::make_timespan(cct->_conf->rgw_fastfail_homeless_timeout), [this, tid]() {_op_cancel(tid, -ETIMEDOUT);});
```

Comment:
i'm not familiar with homeless sessions or how they transition to/from normal sessions, but something needs to cancel this timer if we get an osd map that allows us to satisfy the Op